server/src/main/java/org/elasticsearch/transport/InboundPipeline.java

```diff
@@ -25,33 +25,39 @@
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.common.lease.Releasable;
 import org.elasticsearch.common.lease.Releasables;
+import org.elasticsearch.common.unit.TimeValue;
 import org.elasticsearch.common.util.PageCacheRecycler;
 
 import java.io.IOException;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 import java.util.function.BiConsumer;
+import java.util.function.LongSupplier;
 
 public class InboundPipeline implements Releasable {
 
     private static final ThreadLocal<ArrayList<Object>> fragmentList = ThreadLocal.withInitial(ArrayList::new);
     private static final InboundMessage PING_MESSAGE = new InboundMessage(null, true);
 
+    private final LongSupplier relativeTimeInMillis = () -> TimeValue.nsecToMSec(System.nanoTime());
```

Comment:
I think we should pass this in via the constructor so that we can continue to use `ThreadPool.relativeTimeInMillis()`?